src/features/user/MFV2/MyContributions/ProjectHeader.tsx

```diff
@@ -0,0 +1,71 @@
+import styles from './MyContributions.module.scss';
+import ProjectTypeIcon from '../../../projects/components/ProjectTypeIcon';
+import { useTranslations } from 'next-intl';
+import { EcosystemTypes, TreeProjectClassification } from '@planet-sdk/common';
+import { useMemo } from 'react';
+
+type ConservationProps = {
+  projectPurpose: 'conservation';
+  projectEcosystem: Exclude<
+    EcosystemTypes,
+    'tropical-forests' | 'temperate'
+  > | null;
+};
+
+type TreesProps = {
+  projectPurpose: 'trees';
+  projectClassification: TreeProjectClassification | null;
+};
+
+type Props = (ConservationProps | TreesProps) & {
+  projectName: string;
+  color?: 'light' | 'dark';
+};
+
+const ProjectHeader = ({ color = 'dark', ...otherProps }: Props) => {
+  const tProject = useTranslations('Project');
+  const { projectName, projectPurpose } = otherProps;
+
+  const projectType =
+    projectPurpose === 'trees'
+      ? otherProps.projectClassification
+      : otherProps.projectPurpose;
```

Comment:
Since you have destructured the otherProps you could use projectPurpose and other properties from there

Reply:
Not possible as type checks on discriminated unions don't work if we destructure the props. Props that aren't common to all the individual types show up with warnings that they are not present on the union of the types.